plugins/com.google.cloud.tools.eclipse.googleapis/src/com/google/cloud/tools/eclipse/googleapis/internal/DefaultAccountProvider.java

```diff
@@ -63,43 +97,86 @@ public void initialize(HttpRequest httpRequest) throws IOException {
    */
   @Override
   public Optional<Account> getAccount(){
+    return INSTANCE.computeAccount();
+  }
+  
+  private Optional<Account> computeAccount() {
+    if (!currentCred.isPresent()) {
+      return Optional.empty();
+    }
+    if (cachedAccount.isPresent()) {
+      return cachedAccount;
+    }
+    Credential credential = currentCred.get();
     try {
-      return Optional.of(getAccount(GoogleCredential.getApplicationDefault()));
+      HttpRequestInitializer chainedInitializer = new HttpRequestInitializer() {
+        @Override
+        public void initialize(HttpRequest httpRequest) throws IOException {
+          credential.initialize(httpRequest);
+          requestTimeoutSetter.initialize(httpRequest);
+        }
+      };
+      
+      Oauth2 oauth2 = new Oauth2.Builder(transport, jsonFactory, credential)
+          .setHttpRequestInitializer(chainedInitializer)
+          .setApplicationName(CloudToolsInfo.USER_AGENT)
+          .build();
+      
+      UserInfo userInfo = new UserInfo(oauth2.userinfo().get().execute());
+      Optional<Account> result = Optional.of(new Account(
+          userInfo.getEmail(), credential, userInfo.getName(), userInfo.getPicture()));
+      cachedAccount = result;
+      return result;
     } catch (IOException ex) {
-      LOGGER.log(Level.SEVERE, "IOException occurred when obtaining ADC", ex);
+      LOGGER.log(Level.SEVERE, "Error when computing account from ADC file", ex);
       return Optional.empty();
     }
-    
   }
   
   /**
    * @return the ADC if set
    */
   @Override
   public Optional<Credential> getCredential() {
-    return getAccount().map(Account::getOAuth2Credential);
+    return currentCred;
   }
   
-  Account getAccount(Credential credential) throws IOException {
-    if (accountCache.containsKey(credential)) {
-      return accountCache.get(credential);
+  private Optional<Credential> computeCredential() {
+    Optional<File> credsFile = getCredentialFile();
+    if (!credsFile.get().exists()) { 
```

Comment:
If getCredentialFile returns Optional.empty(), this get() call would throw a NullPointerException.

If getCredentialFile will always return a File object (which, as implemented now, it will), then it doesn't need to be wrapped in an Optional.

Reply:
I removed the `Optional` wrapper.